### PR TITLE
`azurerm_eventgrid_domain` SKU support

### PR DESCRIPTION
**Pull request by jrauschenbusch**
Adds SKU support to `azurerm_eventgrid_domain` resource.

### CODE REVIEW

File: azurerm/internal/services/eventgrid/tests/eventgrid_domain_resource_test.go [hunk 1]
```diff
@@ -250,3 +270,26 @@ resource "azurerm_eventgrid_domain" "test" {
...
+  name     = "acctestRG-%d"
+  location = "%s"
+}
+
+resource "azurerm_eventgrid_domain" "test" {
+  name                = "acctesteg-%d"
+  location            = azurerm_resource_group.test.location
+  resource_group_name = azurerm_resource_group.test.name
+
+  sku {
```
Comment (mbfrahry) [inline azurerm/internal/services/eventgrid/tests/eventgrid_domain_resource_test.go:290]: We'll want to change this around a bit.
```suggestion
  sku = "Premium"
```

**Comment by jrauschenbusch** [comment]
> Hey @jrauschenbusch, this looks good minus one comment but Premium skus don't seem to be supported yet.
> 
> ```
> "Create or update to Premium sku is currently not supported."
> ```
> 
> Were you able to get Premium skus working on your end?

Hey @mbfrahry,

you're right. This is a preview feature and is not GA. As there was no stable EventGrid API update for a longer period we upgraded to the newest preview API version. I'll close this PR. Maybe it will become relevant again within the near future.

Coming with `azure-sdk-for-go` v43.2.0 there is now also a new stable release:
https://github.com/Azure/azure-sdk-for-go/tree/master/services/eventgrid/mgmt/2020-06-01

I'll use the models of this API to figure out which extensions are already available (for my next PRs).

Sorry for the circumstances!

File: azurerm/internal/services/eventgrid/eventgrid_domain_resource.go [hunk 2]
```diff
@@ -52,6 +53,18 @@ func resourceArmEventGridDomain() *schema.Resource {
...
+			"sku": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  string(eventgrid.Basic),
+				ForceNew: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					string(eventgrid.Basic),
+					string(eventgrid.Premium),
+				}, false),
+				DiffSuppressFunc: suppress.CaseDifference,
```
Comment (tombuildsstuff) [inline azurerm/internal/services/eventgrid/eventgrid_domain_resource.go:65]: can we make this case sensitive?

```suggestion
```

**Comment by jrauschenbusch** [comment]
@tombuildsstuff @mbfrahry is right, there is currently no official support for changing the SKU. Also with the latest stable API (`2020-06-01`) the feature was not officially released. I'll close this PR and reopen it if it's getting released.

**Pull request closed without merging**